client/reader/stream/index.jsx

```diff
@@ -190,6 +190,12 @@ class ReaderStream extends Component {
 			window.history.scrollRestoration = 'manual';
 		}
 
+		if ( this.props.selectedPostKey ) {
+			setTimeout( () => {
+				this.focusSelectedPost( this.props.selectedPostKey );
+			}, 100 );
+		}
```

Comment:
Actually one thing that just occurred to me: does it make sense to reuse the existing `componentDidUpdate()` behavior:

https://github.com/Automattic/wp-calypso/blob/323f51b38fad7e34634b2a2eca7d9eeff8d1baea/client/reader/stream/index.jsx#L126-L129

and ideally, have a single approach for focusing the post on load and on update?

Reply:
I don't understand. I don't think we can reuse it

* We can't use the condition `keysAreEqual( selectedPostKey, this.props.selectedPostKey )` in `componentDidMount` because `{ selectedPostKey }` is an argument to `componentDidUpdate` and unavailable in `componentDidMount`.
* The scroll restoration seems to be handled differently already and depends on `if ( 'scrollRestoration' in window.history )`
* and we need the `setTimeout` here because DOM isn't fully ready yet, whereas it is not neccessary in the `componentDidUpdate` function

Reply:
Makes sense, thanks for confirming! 🚀